standalone/pruntime/pruntime/src/pal_gramine.rs

```diff
@@ -0,0 +1,136 @@
+use log::info;
+use std::alloc::System;
+
+use phactory_pal::{Machine, MemoryStats, MemoryUsage, ProtectedFileSystem, Sealing, RA};
+use phala_allocator::StatSizeAllocator;
+use std::fs::File;
+
+use crate::ra;
+
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
+pub(crate) struct GraminePlatform;
+
+impl Sealing for GraminePlatform {
+    type SealError = anyhow::Error;
+    type UnsealError = anyhow::Error;
+
+    fn seal_data(
+        &self,
+        path: impl AsRef<std::path::Path>,
+        data: &[u8],
+    ) -> Result<(), Self::SealError> {
+        std::fs::write(path, data)?;
+        Ok(())
+    }
+
+    fn unseal_data(
+        &self,
+        path: impl AsRef<std::path::Path>,
+    ) -> Result<Option<Vec<u8>>, Self::UnsealError> {
+        match std::fs::read(path) {
+            Ok(data) => Ok(Some(data)),
+            Err(err) => match err.kind() {
+                std::io::ErrorKind::NotFound => Ok(None),
+                _ => Err(err.into()),
+            },
+        }
+    }
+}
+
+pub struct ProtectedFile(File);
+
+impl std::io::Read for ProtectedFile {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        self.0.read(buf)
+    }
+}
+
+impl std::io::Write for ProtectedFile {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.0.write(buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        self.0.flush()
+    }
+}
+
+impl ProtectedFileSystem for GraminePlatform {
+    type IoError = std::io::Error;
+
+    type ReadFile = ProtectedFile;
+
+    type WriteFile = ProtectedFile;
+
+    fn open_protected_file(
+        &self,
+        _path: impl AsRef<std::path::Path>,
+        _key: &[u8],
+    ) -> Result<Option<Self::ReadFile>, Self::IoError> {
+        unimplemented!()
+    }
+
+    fn create_protected_file(
+        &self,
+        _path: impl AsRef<std::path::Path>,
+        _key: &[u8],
+    ) -> Result<Self::WriteFile, Self::IoError> {
+        unimplemented!()
+    }
+}
+
+impl RA for GraminePlatform {
+    type Error = anyhow::Error;
+
+    fn create_attestation_report(
+        &self,
+        data: &[u8],
+    ) -> Result<(String, String, String), Self::Error> {
+        // TODO.kevin: move the key out of the binary?
```

Comment:
Not a big deal. We will move to DCAP where the key is not necessary anyway in the future